.github/workflows/build.yml

```diff
@@ -86,13 +86,30 @@ jobs:
         id: west-zephyr-export
         with:
           args: "zephyr-export"
+      - name: Prepare build parameters
+        id: build-params
+        run: |
+          SHIELD=
+          if [ -n "${{ matrix.shield }}" ]; then
+            SHIELD="-DSHIELD=${{ matrix.shield }}"
+          fi
+          echo ::set-output name=shield::${SHIELD}
```

Comment:
Thoughts on using something more generic, to allow other build steps to optionally add other defines?

Reply:
Yeah, that's a good question and that's actually what I did for `zmk-docker`'s workflow (which is more refined).

I suspect in this case I was just powering through and didn't take a step back before pushing.

I can see the merits so I'll refactor it and re-push.

Reply:
That seems to have done the trick.  This build (a PR which will build off this one) demonstrates the result:
https://github.com/zmkfirmware/zmk/runs/1501689317